docker/compose/gen_docker_compose.py

```diff
@@ -0,0 +1,157 @@
+
+    # parse the config yaml file to get vehicle names
+    config_dict = None
+    with open(args.config_file, "r") as stream:
+        try:
+            config_dict = yaml.safe_load(stream)
+        except yaml.YAMLError as ex:
+            print(ex)
+
+    template = """
+    version: '2.4'
+    services:
+      sim:
+        image: cloudsim_sim
+        command: world:={{ world }} config_file:=/home/developer/config/config.yaml sim_mode:=sim headless:={{ headless }}
+        networks:
+          sim_net:
+            ipv4_address: 172.28.1.1
+        environment:
+          - DISPLAY
+          - QT_X11_NO_MITSHM=1
+          - XAUTHORITY=/tmp/.docker.xauth
+          - IGN_PARTITION=sim
+          - IGN_IP=172.28.1.1
+          - ROS_LOCALHOST_ONLY=1
+          - ROS_DOMAIN_ID=99
+        privileged: true
+        runtime: nvidia
+        security_opt:
+          - seccomp=unconfined
+        volumes:
+          - "/tmp/.docker.xauth:/tmp/.docker.xauth"
+          - "/tmp/.X11-unix:/tmp/.X11-unix"
+          - "/etc/localtime:/etc/localtime:ro"
+          - "/dev/input:/dev/input"
+          - "./:/home/developer/config/"
+      {% for robot in config_dict -%}
+      bridge{{ loop.index }}:
+        image: cloudsim_bridge
+        command: world:={{ world }} config_file:=/home/developer/config/config.yaml sim_mode:=bridge robot:={{ robot["model_name"] }}
+        networks:
+          relay_net{{ loop.index }}:
+            ipv4_address: 172.{{29 + loop.index - 1 }}.1.1
+          sim_net:
+            ipv4_address: 172.28.1.{{ loop.index + 1 }}
+        environment:
+          - IGN_PARTITION=sim
+          - IGN_IP=172.28.1.{{ loop.index + 1 }}
+          - ROS_DOMAIN_ID={{ loop.index }}
+          - CYCLONEDDS_URI=file:///home/developer/config/cyclonedds.xml
+        volumes:
+          - "./:/home/developer/config/"
+        depends_on:
+          - "sim"
+      solution{{ loop.index }}:
+        image: {{ image }}
+        command: {{ robot["model_name"] }}
+        # uncommment to enable interactive shell
+        # stdin_open: true
+        # tty: true
+        networks:
+          relay_net{{ loop.index }}:
+            ipv4_address: 172.{{ 29 + loop.index - 1 }}.1.2
+        runtime: nvidia
+        environment:
+          - ROS_DOMAIN_ID={{ loop.index }}
+        depends_on:
+          - "bridge{{ loop.index }}"
+      {% endfor %}
+    networks:
+      sim_net:
+        ipam:
+          driver: default
+          config:
+            - subnet: 172.28.0.0/16
+      {% for robot in config_dict -%}
+      relay_net{{ loop.index }}:
+        # internal: true
+        ipam:
+          driver: default
+          config:
+            - subnet: 172.{{ 29 + loop.index - 1 }}.0.0/16
+      {% endfor %}
+    """
+
+    data = { "image": args.image,
+             "config_dict": config_dict,
+             "world": args.world,
+             "headless": 'True' if args.headless else 'False',
```

Comment:
This should work.

```suggestion
             "headless": str(args.headless),
```

Reply:
I kept this the same since we're now doing string comparison against `true`.